actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -480,6 +487,8 @@ def []=(name, options)
           options = { value: options }
         end
 
+        preprocess_options(options)
```

Comment:
If I'm following this code correctly, I think calling `@parent_jar[name] = options` here will lead to `preprocess_options` being called again on line 398 in the parent jar. Thus, maybe we only need to call it once in the parent jar itself?

You are right that we do need the options to be "converted" before the cookie is actually written.